typings.d.ts

```diff
@@ -0,0 +1,80 @@
+declare module 'gavel' {
+  export enum RESTMethods {
+    GET = 'GET',
+    POST = 'POST',
+    PUT = 'PUT',
+    PATCH = 'PATCH',
+    DELETE = 'DELETE',
+    OPTIONS = 'OPTIONS'
+  }
+
+  export interface HTTPMessage {
+    method: RESTMethods;
+    uri: string;
+    statusCode: string | number;
+    headers: Record<string, string> | string;
+    body: Record<string, any> | string;
+    /**
+     * [JSON Schema](https://json-schema.org/) Draft V3-4.
+     */
+    bodySchema: Record<string, any> | string;
```

Comment:
This is more of a question about TypeScript than PR comment.

Context: In the future JSON Schema Draft 6, booleans will also valid schema anywhere. Whereas in Draft 4 they are only permittable in 'additionalProperties' and 'additionalItems' properties. [(true schema validates like a `{}` schema, false schema validates like `{"not": {}}`)](https://github.com/kylef/JSONSchema.swift/blob/302a7dbb4f3a85cdb8afa5632b7058572a60eea0/Sources/Draft4Validator.swift#L48). 

Question: Will adapting this type definition to support JSON Schema Draft 6 become a breaking change to gavel.js as this is part of the interface/contract to library consumers. Which would cause major version of gavel.js?

Reply:
I would put this question other way: will we deprecate JSON Schema Draft > 5 when introducing Draft V6+?

If not, the proper way would be to use a union type:

```ts
interface HTTPMessage {
  bodySchema: JSONSchemaV6 | JSONSchemaV5 | JSONSchemaV4
}
```

Types derive from the implementation, so if Gavel supports all those JSON Schema draft versions, then there is no breaking change. If we decide to deprecate older versions support, that would be a breaking change by definition, since the actual code of the library will change, not only types.